api/search/pathparser.py

```diff
@@ -0,0 +1,98 @@
+"""
+The purpose of this module is to parse a query path
+(expected in a xpath-like format).
+we use only "/" and "//" to indicate respectively children and generic descendants.
+
+Examples:
+"collections/sessions/acquisitions/files" and "collections//acquisitions/files"
+will prepare a search for files in an acquisition in a collection
+
+"collections//files"
+a search for files in collections, sessions and acquisitions (belonging to a collections)
+
+"acquisitions" and "projects//acquisitions"
+a search for acquisitions (as incidentally every acquisition is included in a project)
+
+"""
+from collections import deque as Queue
```

Comment:
Just declaring my intention of using it as a queue.

I can use the deque name if we fear name conflicts.